include/swift/AST/DiagnosticGroups.def

```diff
@@ -72,6 +72,7 @@ GROUP(TemporaryPointers, "temporary-pointers")
 GROUP(TrailingClosureMatching, "trailing-closure-matching")
 GROUP(UnknownWarningGroup, "unknown-warning-group")
 GROUP(CompilationCaching, "compilation-caching")
+GROUP(WeakMutability, "weak-mutability")
```

Comment:
There isn't an existing group for the `var`-isn't-mutated warning, right? We don't need to introduce that right away, but it'd be nice if this wasn't inconsistent with a reasonable choice.

Have you talked about this on the forums? I think the process is to start a thread in Development > Compiler.

Reply:
> There isn't an existing group for the var-isn't-mutated warning, right? 

Yes. Original warnings `variable_never_mutated` and `variable_tuple_elt_never_mutated` are not part of a group.

> Have you talked about this on the forums?

Not yet.

> I think the process is to start a thread in Development > Compiler.

Where is this process described? I don't see any mentioning of the process in the [SE-0443](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0443-warning-control-flags.md). Would be nice to make this more visible.

Started - https://forums.swift.org/t/naming-warnings-group-for-variable-mutability/81356

Reply:
I'm fine with this name. We can always revise it if there's more feedback in the thread.